apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/SurveyAnalysisCTA.tsx

```diff
@@ -102,8 +106,17 @@ export const SurveyAnalysisCTA = ({
     setLoading(false);
   };
 
-  const getPreviewUrl = () => {
+  const getPreviewUrl = async () => {
     const separator = surveyUrl.includes("?") ? "&" : "?";
```

Comment:
Can we replace the separator logic here with actually manipulating URLs? then we can simply do something like `surveyUrl.searchParams.set("suid", newId)`.